src/redux/books/booksSlice.js

```diff
@@ -1,19 +1,64 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
 import { v4 as uuidv4 } from 'uuid';
-import bookItems from '../bookItems';
 
 const initialState = {
-  booksList: bookItems,
+  booksList: [],
+  isLoading: false,
 };
 
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/ENfILofDRr0lFes8m9WT/books';
+export const getBookItems = createAsyncThunk('books/getBookItems', async () => {
+  let response = []; // Declare response as an empty array initially
+
+  try {
+    const resp = await axios(url);
+    response = Object.entries(resp.data).map(([key, value]) => ({ item_id: key, ...value }));
+    return response;
+  } catch (error) {
+    console.log(error);
+    return error;
+  }
+});
+
+const addBookurl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/ENfILofDRr0lFes8m9WT/books';
+export const postBook = createAsyncThunk('books/postBookItems', async ({ title, author }) => {
+  const newBook = {
+    item_id: uuidv4(),
+    title,
+    author,
+    category: 'Fiction',
+    percentage: 1,
+    chapter: 34,
+  };
+  try {
+    const resp = await axios.post(addBookurl, newBook);
+    return resp.data;
+  } catch (error) {
+    console.log(error);
+    return error;
```

Comment:
- [Optional] Kindly, Do not commit `console.log` to your repo. It’s ugly, it kills performance and it can make confidential data be visible to anyone using the browser tools to look at your website. check the [js best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) for more info 👍